File: Labs/projext/app/src/main/java/com/example/cosereaionutc114a/MainActivity.java
package com.example.cosereaionutc114a;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.ListView;

import androidx.appcompat.app.AppCompatActivity;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private static final int REQUEST_CODE_ADD_HT = 200;
    private ArrayList<HusaTelefon> htList;
    private Button btnOpenForm;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        htList = new ArrayList<>();

        ListView listView = findViewById(R.id.coserea_ionut_listViewHusa);
        btnOpenForm = findViewById(R.id.coserea_ionut_btnOpenForm);

        // Adaugare obiect nou
        btnOpenForm.setOnClickListener(v -> {
            Intent intent = new Intent(MainActivity.this, NewHusa.class);
            startActivityForResult(intent, REQUEST_CODE_ADD_HT);
        });
    }

    // Preluare rezultat din AddCiActivity (adaugare sau modificare)
    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);

        if (requestCode == REQUEST_CODE_ADD_HT && resultCode == RESULT_OK && data != null) {
            HusaTelefon ht = data.getParcelableExtra("ht");
            int position = data.getIntExtra("position", -1);

            if (position == -1) {
                // Obiect nou
                htList.add(ht);
            } else {
                // Modificare obiect existent
                htList.set(position, ht);
            }
        }
    }
}

File: Labs/Lab4/app/src/main/java/com/example/lab4/MainActivity.kt
package com.example.lab4

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {

    private val requestCodeSecondActivity = 1
    private lateinit var textViewCarte: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        val buttonAdaugaCarte = findViewById<Button>(R.id.buttonAdaugaCarte)
        textViewCarte = findViewById(R.id.textViewCarte)

        buttonAdaugaCarte.setOnClickListener {
            val intent = Intent(this, Lab4SecondActivity::class.java)
            startActivityForResult(intent, requestCodeSecondActivity)
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == requestCodeSecondActivity && resultCode == Activity.RESULT_OK) {
            val carte: Carte? = data?.getParcelableExtra("carte")

            carte?.let {
                val carteInfo = """
                    Titlu: ${it.titlu}
                    Autor: ${it.autor}
                    An Publicatie: ${it.anPublicatie}
                    Numar Pagini: ${it.pagini}
                    Este eBook: ${if (it.esteEbook) "Da" else "Nu"}
                    Gen: ${it.gen.name}
                """.trimIndent()

                textViewCarte.text = carteInfo
            }
        }
    }
}
